src/modules/flow/netctl/netctl.c

```diff
@@ -0,0 +1,327 @@
+    }
+
+    mdata->state = current_state;
+
+    return;
+
+error:
+    mdata->state = current_state;
+
+    SOL_WRN("service %s failed to send online state", name);
+}
+
+void
+error_cb(void *data, const struct sol_netctl_service *service,
+    unsigned int error)
+{
+    struct sol_flow_node *node = data;
+    struct network_service_data *mdata = sol_flow_node_get_private_data(node);
+    const char *name;
+
+    name = sol_netctl_service_get_name(service);
+
+    if (!name || strcmp(name, mdata->service_name))
+        return;
+
+    sol_flow_send_error_packet(node, -error, "Service %s error is: %s",
+        name, sol_util_strerrora(-error));
+}
+
+static int
+connect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    struct sol_netctl_service *service;
+    int r;
+
+    service = sol_netctl_find_service_by_name(mdata->service_name);
+    if (!service) {
+        sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+        return 0;
+    }
+
+    r = sol_netctl_service_connect(service);
+    if (r < 0)
+        sol_flow_send_error_packet(node, -r, "Could not connect to service: %s: %s",
+            mdata->service_name, sol_util_strerrora(-r));
+
+    return 0;
+}
+
+static int
+disconnect_service_name(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct network_service_data *mdata = data;
+    struct sol_netctl_service *service;
+    int r;
+
+    service = sol_netctl_find_service_by_name(mdata->service_name);
+    if (!service) {
+        sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+        return 0;
+    }
+
+    r = sol_netctl_service_disconnect(service);
+    if (r < 0)
+        sol_flow_send_error_packet(node, -r, "Could not disconnect to service: %s: %s",
+            mdata->service_name, sol_util_strerrora(-r));
+
+    return 0;
+}
+
+static int
+state_dispatch(struct sol_flow_node *node, void *data)
+{
+    struct network_service_data *mdata = data;
+    struct sol_netctl_service *service;
+    int r;
+
+    service = sol_netctl_find_service_by_name(mdata->service_name);
+    if (!service) {
+        sol_flow_send_error_packet(node, ENOENT, "Did not found service name %s", mdata->service_name);
+        return 0;
+    }
+    r = sol_flow_send_string_packet(node,
+        SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__OUT,
+        sol_netctl_service_state_to_str(mdata->state));
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (mdata->state == SOL_NETCTL_SERVICE_STATE_ONLINE)
+        r = sol_flow_send_bool_packet(node,
+            SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+            true);
+    else
+        r = sol_flow_send_bool_packet(node,
+            SOL_FLOW_NODE_TYPE_NETCTL_SERVICE__OUT__ONLINE,
+            false);
```

Comment:
why not a single `send_bool_packet()` with the last argument being: `mdata->state == SOL_NETCTL_SERVICE_STATE_ONLINE`? -